### DISCUSSION TITLE
auto-deleting old version when new version is installed

### DISCUSSION THREAD

**Discussion by Amogh-Gurudatta** [ThioJoe/YT-Spammer-Purge#328] [Ideas]
When a new version is automatically installed, the app should be able to delete the old version automatically. Keeping it would be useless, as the newer version has more fixes and/ or features. This can be done after the install is finished, or when the new version is run. Maybe [this](https://stackoverflow.com/questions/10112601/how-to-make-scripts-auto-delete-at-the-end-of-execution) would be helpful.

**Comment by kiatchun** [comment]
YES!

**Reply by Kurdakov** [reply]
but client_secrets.json should be preserved (or copied)

**Reply by Amogh-Gurudatta** [reply]
only the part(s) of the app should be deleted, not cliient_secrets or token.pickle.

**Comment by KendallDoesCoding** [comment]
> 

 This is a good idea. It can also print out a message saying "You have the newest version of the YT-Spammer-Purger app, Do you want to un-install the previous one [Y] [N]"

**Comment by ThioJoe** [comment]
I don't think this would be a good idea as default. But maybe it could be a config setting. Idk if the exe could even delete itself.

**Reply by Amogh-Gurudatta** [reply]
ig a config setting works, but on by default.

**Reply by Amogh-Gurudatta** [reply]
Also, the file need not delete itself. The newer version can check to see if an older version is present and delete that.

**Reply by rbeer** [reply]
> Idk if the exe could even delete itself.

This is a good opportunity to start the refactoring, regarding the ~4k lines in one file.
This is a prime example for the UNIX philosophy I mentioned in https://github.com/ThioJoe/YT-Spammer-Purge/discussions/84#discussioncomment-1697762 :
`Make each program do one thing well. To do a new job, build afresh rather than complicate old programs by adding new "features".`

Updating is an entirely different task/feature from the program's main purpose of working with YouTube comments.
And let's be honest, you thought about that, didn't you? I saw your "future proof" comments in the updater code. :wink:

I'll come up with a solution and PR, if you don't have any veto.

**Reply by ThioJoe** [reply]
Well I'm just going to say outright I don't want the program deleting old exe's so i'm not going to implement that.

**Comment by jac0-b** [comment]
Definitely worth checking out [innosetup](https://jrsoftware.org/isinfo.php) for a smoother installation process. I use it for my project AlbumPaper and it works great :)

**Reply by ThioJoe** [reply]
I use pyinstaller and already have the config set up for it, as well as a script that handles every step of the release process, so I'm just going to stick with that

**Reply by jac0-b** [reply]
Fair enough if it ain't broke don't fix it. Innosetup is just for after you have to exe (after pyinstaller). For a simple project like this the only advantages I could think of is giving a nice GUI for picking an installation location and adding a start menu icon and to programs and files for easy uninstall so it may not be worth the effort if you're unfamiliar with it.
